OnlineStore/store/src/main/java/product/ListProductComparators.java

```diff
@@ -0,0 +1,56 @@
+package product;
+
+import product.Product;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+public class ListProductComparators {
+    Map<String, String> sortConditions;
+    //private List<Comparator<Product>> comparatorList;
+
+    public ListProductComparators(Map<String, String> sortConditions) {
+        this.sortConditions = sortConditions;
+    }
+
+    public List<Comparator<Product>> getComparatorList(Map<String, String> sortConditions) {
+        List<Comparator<Product>> comparatorList = new ArrayList<>();
+        for (Map.Entry<String, String> item : sortConditions.entrySet()) {
+            String key = item.getKey().trim().toLowerCase();
+            String value = item.getValue().trim().toLowerCase();
+            switch (key) {
+                case "name":
+                    if (value.equals("asc")) {
+                        comparatorList.add(Comparator.comparing(Product::getName));
+                    } else if (value.equals("desc")) {
```

Comment:
im not shure. but i think there  is a method like descending or reversOrder in comparator, and you can siplify your solution